### PR TITLE
Cycle detection

### LINKED ISSUE #20: Prevent cycles in graph traversal
Currently it's possible to create cyclic graph. We need to prevent it. Error message should be easily visible for User (in logs?)

### PR DESCRIPTION
**Pull request by nebril**
fixes #20

### CODE REVIEW

File: cmd/kubeac/main.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ func main() {
 	depGraph, err := scheduler.BuildDependencyGraph(c, sel)
 	if err != nil {
 		log.Fatal(err)
+		return
```
Comment (gluke77) [inline cmd/kubeac/main.go]: log.Fatal stops execution. we dont need 'return' here.

File: scheduler/scheduler.go [hunk 2]
```diff
@@ -149,9 +150,20 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (*DependencyG
 			depGraph[d.Parent].RequiredBy, depGraph[d.Child])
 	}
 
-	return &depGraph, nil
+	cycles := DetectCycles(depGraph)
```
Comment (gluke77) [inline scheduler/scheduler.go]: I'd move all this cycles part to main(), between building and executing depgraph. Now BuildDependencyGraph func is already too big'n'complex. I'm thinking how to refactor it a bit.

Comment (nebril) [inline scheduler/scheduler.go]: Yeah, that was also my impression too, but I thought that cycle detection is too low-level operation for main func.

File: cmd/kubeac/main.go (hunk 1 above)
Comment (nebril) [inline cmd/kubeac/main.go]: Cool, will remove

File: scheduler/scheduler.go (hunk 2 above)
Comment (gluke77) [inline scheduler/scheduler.go]: > but I thought that cycle detection is too low-level operation
not really, as for my taste )). Also, look at yr tests -- they definitely look like this is a separate part ,)

Comment (gluke77) [inline scheduler/scheduler.go]: Apart from the above LGTM. Test coverage is awesome!!

Comment (nebril) [inline scheduler/scheduler.go]: Thanks! Having the test boilerplate really helped. I will move the check to main func.

**Pull request merged**